### PR TITLE
readd existing members to a conversation should not fail

### PR DESCRIPTION
**Pull request by AMarcedone**

### CODE REVIEW

File: go/chat/server.go [hunk 1]
```diff
@@ -2536,12 +2536,44 @@ func (h *Server) BulkAddToConv(ctx context.Context, arg chat1.BulkAddToConvArg)
...
+	sender := NewBlockingSender(h.G(), boxer, h.remoteClient)
+
+	usernamesToAdd := arg.Usernames
+	toExclude := make(map[keybase1.UID]bool)
+
+	// retry the add a few times to prevent races. Each time we remove members that are already part of the conversation.
+	for i := 0; i < 4 && len(usernamesToAdd) > 0; i++ {
+		h.Debug(ctx, "BulkAddToConv: trying to add %v", usernamesToAdd)
+		err = sendBulkAddToConv(ctx, sender, usernamesToAdd, arg.ConvID, info)
+		if err == nil {
```
Comment (joshblum) [inline go/chat/server.go]: prefer `switch err := err.(type)` here

File: go/libkb/errors.go [hunk 2]
```diff
@@ -2131,6 +2131,17 @@ func (e ChatClientError) IsImmediateFail() (chat1.OutboxErrorType, bool) {
 
 //=============================================================================
 
+type ChatUsersAlreadyInConversationError struct {
+	Uids   []keybase1.UID
+	ConvID chat1.ConversationID
```
Comment (joshblum) [inline go/libkb/errors.go]: what is this used for?

File: go/libkb/rpc_exim.go [hunk 3]
```diff
@@ -601,6 +601,17 @@ func ImportStatusAsError(g *GlobalContext, s *keybase1.Status) error {
 		}
 	case SCChatClientError:
 		return ChatClientError{Msg: s.Desc}
+	case SCChatUsersAlreadyInConversationError:
+		var cid chat1.ConversationID
+		var uids []keybase1.UID
+		for _, field := range s.Fields {
+			if field.Key == "conv_id" {
+				cid = chat1.ConversationID(field.Value)
```
Comment (joshblum) [inline go/libkb/rpc_exim.go]: need to call `MakeConvID` to hex decode here, but I think you can just drop this field altogether

File: go/libkb/errors.go (hunk 2 above)
Comment (AMarcedone) [reply go/libkb/errors.go]: I'll drop it. thought it would be useful for logging/debug purposes as it.

**Comment by AMarcedone** [comment]
fixed.

**Comment by AMarcedone** [comment]
Of course. I will merge this first so I can revendor out of master.

**Pull request merged**